src/diet/index.ts

```diff
@@ -0,0 +1,506 @@
+
+    const baseCost = menuItem.price() + sum(helpers, (item) => item.price());
+    const valueRaw =
+      expectedAdventures(menuItem.item, defaultModifiers) * this.mpa -
+      baseCost +
+      (menuItem.additionalValue ?? 0);
+    const valueForkMug =
+      expectedAdventures(menuItem.item, {
+        ...defaultModifiers,
+        forkMug: true,
+      }) *
+        this.mpa -
+      baseCost -
+      forkMugPrice +
+      (menuItem.additionalValue ?? 0);
+
+    const valueSpleen = $items`jar of fermented pickle juice, extra-greasy slider`.includes(
+      menuItem.item
+    )
+      ? 5 * this.spleenValue
+      : 0;
+
+    return forkMug && valueForkMug > valueRaw
+      ? [[...helpers, forkMug, menuItem], valueForkMug + valueSpleen]
+      : [[...helpers, menuItem], valueRaw + valueSpleen];
+  }
+
+  planOrgan(
+    organ: Organ,
+    capacity: number,
+    overrideModifiers: Partial<ConsumptionModifiers> = {}
+  ): [number, [MenuItem[], number][]] {
+    // print(`Plan ${organ} < ${capacity}`);
+    const submenu = this.menu.filter((item) => item.organ === organ);
+    const knapsackValues = submenu.map(
+      (menuItem) =>
+        [
+          ...this.consumptionHelpersAndValue(menuItem, overrideModifiers),
+          menuItem.size,
+          menuItem.maximum,
+        ] as [MenuItem[], number, number, number?]
+    );
+    return knapsack(knapsackValues, capacity);
+
+    // print(
+    //   `Items: ${itemList.length} ${([] as Item[])
+    //     .concat(...itemList)
+    //     .map((item) => item.name)
+    //     .join(", ")}`
+    // );
+  }
+
+  planOrgans(
+    organCapacities: OrganSize[],
+    overrideModifiers: Partial<ConsumptionModifiers> = {}
+  ): [number, [MenuItem[], number][]] {
+    const valuePlans = organCapacities.map(([organ, capacity]) =>
+      this.planOrgan(organ, capacity, overrideModifiers)
+    );
+    return [
+      sum(valuePlans, ([value]) => value),
+      ([] as [MenuItem[], number][]).concat(
+        ...valuePlans.map(([, plan]) => plan)
+      ),
+    ];
+  }
+
+  planOrgansWithTrials(
+    organCapacities: OrganSize[],
+    trialItems: [MenuItem, OrganSize[]][],
+    overrideModifiers: Partial<ConsumptionModifiers> = {}
+  ): [number, [MenuItem[], number][]] {
+    if (trialItems.length === 0) {
+      return this.planOrgans(organCapacities, overrideModifiers);
+    }
+
+    const organCapacitiesWithMap = new Map(organCapacities);
+    const [trialItem, organSizes] = trialItems[0];
+    // print(`TRYING ${trialItem.item.name}`);
+    for (const [organ, size] of organSizes) {
+      const current = organCapacitiesWithMap.get(organ);
+      if (current !== undefined) {
+        organCapacitiesWithMap.set(organ, current - size);
+      }
+    }
+    const organCapacitiesWith = [...organCapacitiesWithMap];
+
+    const isRefinedPalate =
+      trialItem.item === $item`pocket wish` &&
+      trialItem.wishEffect === $effect`Refined Palate`;
```

Comment:
I do not currently see any code that accounts for already having this active, which is something that can plausibly happen when mushroom wine prices are low--you nightcap with a sacramento+mushroom, and then start the day with it. Similarly, it'd be neat if there was a way to account for "having this buff left over is valuable", but that seems hard and I can't identify what precisely I'd want it to do.

Reply:
yep sounds good